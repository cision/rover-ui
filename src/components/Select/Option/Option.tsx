import React, { useEffect /*, useMemo */ } from 'react';

// import classNames from 'classnames';

import Dropdown from '../../Dropdown';
// import styles from './Option.module.css';

export interface OptionProps {
  children: React.ReactNode;
  // className?: string;
  disabled?: boolean;
  // onClick?: (arg0?: React.MouseEvent<HTMLButtonElement>) => void;
  // selectedValue?: string;
  value?: string;
}

export const Option: React.FC<OptionProps> = ({
  children = null,
  // className = '',
  // disabled = undefined,
  // selectedValue,
  value: customValue,
  ...props
}) => {
  useEffect(() => {
    if (typeof children !== 'string' && !customValue) {
      console.warn(
        'Option requires a `value` prop when `children` is not a string.'
      );
    }
  }, [children, customValue]);

  // const value = useMemo(
  //   () => customValue || (typeof children === 'string' && children) || '',
  //   [children, customValue]
  // );

  return (
    <Dropdown.Menu.Item
      {
        ...{} /*
        disabled={disabled}
        className={classNames(styles.Option, className, {
          [styles.disabled]: disabled,
          [styles.selected]: selectedValue === value,
        })}
        */
      }
      {...props}
    >
      {children}
    </Dropdown.Menu.Item>
  );
};

export default Option;
