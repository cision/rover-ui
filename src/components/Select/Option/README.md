# \<Option\>

The `Option` component is only to be used as a direct child of `Select`.

The only requirement is that it have a string value for either the `children` or `value` prop, since the string is used as the value of the `Select` when the option is selected.

You may optionally pass a `disabled` prop.
