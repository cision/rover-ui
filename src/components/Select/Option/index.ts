export { default } from './Option';
export type { OptionProps } from './Option';
