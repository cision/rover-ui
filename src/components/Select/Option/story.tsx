import React from 'react';

import { storiesOf } from '@storybook/react';
import { boolean, text } from '@storybook/addon-knobs';

import { Wrap } from '../../../stories/storybook-helpers';

import Option from '.';
import Readme from './README.md';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const FancySpan = (props: Record<string, any>) => <strong {...props} />;

storiesOf('Planets/Select/Option', module)
  .addParameters({
    readme: {
      sidebar: Readme,
    },
  })
  .add(
    'Overview',
    () => {
      const name = text('children', 'Option 1');

      return (
        <Wrap>
          <Option
            disabled={boolean('disabled', false)}
            value={text('value', 'custom value')}
          >
            {name}
          </Option>
        </Wrap>
      );
    },
    {
      info: {
        inline: true,
        source: true,
      },
    }
  )
  .add(
    'Examples',
    () => {
      const now = new Date();

      return (
        <Wrap>
          <Option value={now.toISOString()}>
            Now (time) with an ISO string value
          </Option>
          <Option>Just plain old text</Option>
          <Option value="no-tags-here">
            Don&apos;t put <strong>HTML</strong> or
            <FancySpan>React components</FancySpan> in your option unless you
            specify a value.
          </Option>
          <Option disabled>Disabled</Option>
        </Wrap>
      );
    },
    {
      info: {
        inline: true,
        source: true,
      },
    }
  );
