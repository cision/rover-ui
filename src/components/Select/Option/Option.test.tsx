import React from 'react';
import { render } from '@testing-library/react';

import Option from './Option';

describe('Option', () => {
  it('renders', () => {
    const { baseElement } = render(<Option>Boom</Option>);
    expect(baseElement).toMatchSnapshot();
  });

  it("warns when there's no string value provided", () => {
    const warn = jest.spyOn(console, 'warn').mockImplementation(() => {});
    render(
      <Option>
        <strong>Some HTML</strong>
      </Option>
    );

    expect(warn).toBeCalledWith(
      'Option requires a `value` prop when `children` is not a string.'
    );
  });
});
