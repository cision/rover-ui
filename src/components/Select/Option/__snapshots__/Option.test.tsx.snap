// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option renders 1`] = `
<body>
  <div>
    <div
      class="Item content Button text md as-text"
    >
      <span>
        Boom
      </span>
    </div>
  </div>
</body>
`;
